src/components/DatePicker/DatePicker.scss

```diff
@@ -169,3 +186,75 @@ $in-range-border-color: #9ca6de;
   padding-bottom: spacing(extra-tight);
   text-align: center;
 }
+
+.newDesignLanguage {
+  .Day {
+    box-shadow: none;
+
+    &:hover {
+      background: var(--p-interactive-hovered);
+      color: var(--p-text-on-interactive);
+    }
+
+    @include focus-ring;
+    // stylelint-disable-next-line selector-max-specificity
+    &:focus:not(:active) {
+      @include focus-ring($style: 'focused');
+      box-shadow: none;
+    }
+  }
+
+  .Day-selected {
+    background: var(--p-interactive);
+    border: none;
+    color: var(--p-text-on-interactive);
+  }
+
+  // stylelint-disable-next-line selector-max-specificity
+  .Day:not(.Day-lastInRange):not(.Day-firstInRange):not(.Day-inRange) {
+    border-radius: spacing(extra-tight);
+  }
+
+  .Day-firstInRange {
+    @include radius-with-focus-ring {
+      border-radius: spacing(extra-tight);
```

Comment:
I'm not making a suggestion on this one because it's part of the larger replace block below.